internal/logs/logs.go

```diff
@@ -0,0 +1,112 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+
+	cloudwatchlogsclient "cloudfront-invalidation-metrics/internal/aws/cloudwatchlogs"
+)
+
+const (
+	// AwsPayloadLimit is the maximum quality for a data-set to contain
+	// before AWS will reject the payload.
+	AwsPayloadLimit = 20
+)
+
+// ClientInterface for pushing metrics to CloudWatch.
+type ClientInterface interface {
+	Verify(ctx context.Context, logGroupName string, logStreamName string) error
+	Flush(ctx context.Context, logGroupName string, logStreamName string, logData types.InputLogEvent) error
+}
+
+// Client for pushing metrics to CloudWatch.
+type Client struct {
+	CloudWatchLogs cloudwatchlogsclient.ClientInterface
+	Namespace      string
+	LogData        []types.InputLogEvent
+	DryRun         bool
+}
+
+//
+
+// New client for pushing metrics to CloudWatch.
+func New(cloudwatchlogs cloudwatchlogsclient.ClientInterface, namespace string, dryRun bool) (*Client, error) {
+	return &Client{
+		CloudWatchLogs: cloudwatchlogs,
+		Namespace:      namespace,
+		DryRun:         dryRun,
+	}, nil
+}
+
+// Verify logs to Client.
+func (c *Client) Verify(ctx context.Context, logGroupName string, logStreamName string) error {
+
+	// check if LogGroup exists
+	logGroups, err := c.CloudWatchLogs.DescribeLogGroups(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
```

Comment:
Might not even need the verify.

Could do this when submitting events:

* https://github.com/skpr/cloudfront-cloudwatchlogs/blob/main/internal/pusher/pusher.go#L104
* https://github.com/skpr/cloudfront-cloudwatchlogs/blob/main/internal/pusher/pusher.go#L121